ui/src/transaction.ts

```diff
@@ -210,11 +204,28 @@ function txData(txc: ContractTransaction, date: number = Date.now()): TxData {
   };
 }
 
+function registerTxUpdateCallback(tx: Tx): void {
+  if (tx.status === TxStatus.AwaitingInclusion) {
+    const provider = svelteStore.get(walletStore).provider;
+    provider.waitForTransaction(tx.hash).then(txReceipt => {
+      store.update((txs: Tx[]) => {
+        return txs.map(tx_ => {
+          if (tx_.hash === tx.hash) {
+            tx_.status = status(txReceipt);
+          }
+          return tx_;
+        });
```

Comment:
Not that it matters much but `find` and in-place mutation is faster
```suggestion
        const tx_ = txs.find(tx_ => tx_.hash === tx.hash)
        if (tx_) {
          tx_.status = status(txReceipt);
        }
        return txs
```